docs/source/clickhouse/schema_design.rst

```diff
@@ -0,0 +1,46 @@
+ClickHouse Schema Design Best Practices
+=======================================
+
+Columns based on dictionary data
+--------------------------------
+
+ClickHouse is a columnar datastore, and at run-time it loads columns on-demand
+based on the columns referenced in the query (both the columns ``SELECT`` ed
+and those part of the ``WHERE`` clause). The ability to store different columns independently
+and not load them for every row for every query is part of the performance advantage that
+ClickHouse provides over a traditional RDBMS (like PostgreSQL).
+
+Commonly, a data schema contains a flexible key:value pair mapping
+(canonically: ``tags`` or ``contexts``) and stores that
+data in a column that contains two ``Nested`` arrays where the first array contains the keys
+of the dictionary and the second array contains the values. This works well when
+your dataset design gives you the ability to filter based on other attributes to a small
+number of rows and the flexibility of completely arbitrary keys and values are a real requirement.
+Often, however, a ClickHouse user is interested in rows that contain a specific tag key or a
+specific ``key=value`` pair. In this case one is often best-served by creating a top-level
+column and "promoting" the data to it [#dupe]_.
```

Comment:
I think this is misleading as a reason. 
Reading this I would conclude that all scenarios where I query something like ``key=value`` should be promoted. That is the majority of the scenarios where we use tags.
Also the ``tag=value`` case does not really need promotion most of the times as it is covered by the bloomfilter index that makes is potentially cheaper than a dedicated column. 

The dedicated column helps us mostly for arrayJoin, when the tag is present on most rows and where the tag column tends to be really large.

- the arrayJoin case is because the bloomfilter index that filters out data does not help there. We have to scan all tags for all rows 
- the case where the tag key is present on most or all rows and the specific tag value is very very common, because the bloomfilter would filter out very little 


So I would rephrase this paragraph and fix the reason why we promote columns (if there is still a reason at all to be fair)